### PR TITLE
sudo_queue_downward_message

### PR DESCRIPTION
**Pull request by pepyakin**
This PR adds a nifty entrypoint for sending a downward message from root/sudo.

### CODE REVIEW

File: runtime/common/src/paras_sudo_wrapper.rs [hunk 1]
```diff
@@ -58,5 +65,21 @@ decl_module! {
 			runtime_parachains::schedule_para_cleanup::<T>(id);
 			Ok(())
 		}
+
+		/// Send a downward message to the given para.
+		///
+		/// The given parachain should exist and the payload should not exceed the preconfigured size
+		/// `config.max_downward_message_size`.
+		#[weight = (1_000, DispatchClass::Operational)]
+		pub fn sudo_queue_downward_message(origin, id: ParaId, payload: Vec<u8>) -> DispatchResult {
```
Comment (shawntabrizi) [inline runtime/common/src/paras_sudo_wrapper.rs:74]: probably isnt right to use the term "sudo" here. A root call can also come from democracy or other systems which generate a root origin. `root_queue_downward_message` would make more sense but i think `queue_downward_message` would be enough?

**Review by shawntabrizi — APPROVED** [review]
I would not have directly used the term sudo here since you really mean `root` origin which could come from multiple places like democracy, but ya thats just a nitpick

**Comment by pepyakin** [comment]
Yeah, I know. That's just what the module is already called.

**Comment by pepyakin** [comment]
Rebased against the current master. The sole commit of this PR remains unchanged.

File: runtime/common/src/paras_sudo_wrapper.rs [hunk 2]
```diff
@@ -16,22 +16,29 @@
...
 
 /// The module's configuration trait.
-pub trait Trait: paras::Trait + dmp::Trait + ump::Trait + hrmp::Trait { }
+pub trait Trait: configuration::Trait + paras::Trait + dmp::Trait + ump::Trait + hrmp::Trait {}
 
 decl_error! {
-	pub enum Error for Module<T: Trait> { }
+	pub enum Error for Module<T: Trait> {
+		/// The specified parachain or parathread is not registered.
+		ParaDoesnExist,
```
Comment (coriolinus) [inline runtime/common/src/paras_sudo_wrapper.rs:37]:
```suggestion
		ParaDoesntExist,
```

File: runtime/common/src/paras_sudo_wrapper.rs [hunk 3]
```diff
@@ -58,5 +65,21 @@ decl_module! {
...
 		}
+
+		/// Send a downward message to the given para.
+		///
+		/// The given parachain should exist and the payload should not exceed the preconfigured size
+		/// `config.max_downward_message_size`.
+		#[weight = (1_000, DispatchClass::Operational)]
+		pub fn sudo_queue_downward_message(origin, id: ParaId, payload: Vec<u8>) -> DispatchResult {
+			ensure_root(origin)?;
+			ensure!(<paras::Module<T>>::is_valid_para(id), Error::<T>::ParaDoesnExist);
```
Comment (coriolinus) [inline runtime/common/src/paras_sudo_wrapper.rs:76]:
```suggestion
			ensure!(<paras::Module<T>>::is_valid_para(id), Error::<T>::ParaDoesntExist);
```

**Review by coriolinus — APPROVED** [review]

**Review by bkchr — APPROVED** [review]

**Pull request merged**